src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs

```diff
@@ -0,0 +1,149 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Core.Features.Model;
+using Microsoft.Health.Dicom.Functions.Update.Models;
+using Microsoft.Health.Operations.Functions.DurableTask;
+
+namespace Microsoft.Health.Dicom.Functions.Update;
+
+public partial class UpdateDurableFunction
+{
+    /// <summary>
+    /// Asynchronously updates list of instances in a study
+    /// </summary>
+    /// <remarks>
+    /// Durable functions are reliable, and their implementations will be executed repeatedly over the lifetime of
+    /// a single instance.
+    /// </remarks>
+    /// <param name="context">The context for the orchestration instance.</param>
+    /// <param name="logger">A diagnostic logger.</param>
+    /// <returns>A task representing the <see cref="UpdateInstancesAsync"/> operation.</returns>
+    /// <exception cref="ArgumentNullException">
+    /// <paramref name="context"/> or <paramref name="logger"/> is <see langword="null"/>.
+    /// </exception>
+    /// <exception cref="FormatException">Orchestration instance ID is invalid.</exception>
+    [FunctionName(nameof(UpdateInstancesAsync))]
+    public async Task UpdateInstancesAsync(
+        [OrchestrationTrigger] IDurableOrchestrationContext context,
+        ILogger logger)
+    {
+        EnsureArg.IsNotNull(context, nameof(context)).ThrowIfInvalidOperationId();
+        logger = context.CreateReplaySafeLogger(EnsureArg.IsNotNull(logger, nameof(logger)));
+
+        UpdateCheckpoint input = context.GetInput<UpdateCheckpoint>();
+
+        // Backfill batching options
+        input.Batching ??= new BatchingOptions
+        {
+            MaxParallelCount = _options.MaxParallelBatches,
+            Size = _options.BatchSize,
+        };
+
+        input.TotalNumberOfStudies ??= input.StudyInstanceUids.Count;
+
+        if (input.StudyInstanceUids.Count > 0)
+        {
+            string studyInstanceUid = input.StudyInstanceUids[0];
+
+            logger.LogInformation("Beginning to get all instances in a study.");
+
+            IReadOnlyList<InstanceFileIdentifier> instanceWatermarks = await context.CallActivityWithRetryAsync<IReadOnlyList<InstanceFileIdentifier>>(
+                nameof(GetInstanceWatermarksInStudyAsync),
+                _options.RetryOptions,
+                new GetInstanceArguments(input.PartitionKey, studyInstanceUid));
+
+            logger.LogInformation("Getting all instances completed {TotalCount}", instanceWatermarks.Count);
+
+            if (instanceWatermarks.Count > 0)
+            {
+                instanceWatermarks = await context.CallActivityWithRetryAsync<IReadOnlyList<InstanceFileIdentifier>>(
+                        nameof(UpdateInstanceWatermarkAsync),
+                        _options.RetryOptions,
+                        new BatchUpdateArguments(input.PartitionKey, instanceWatermarks, input.ChangeDataset));
+            }
+
+            if (instanceWatermarks.Count > 0)
```

Comment:
I thought about that while development, since I was doing batching I didnt go with that logic. Now I removed the batching in DB, seems like we get rid of the GetWatermark activity.